sycl/include/CL/sycl/detail/helpers.hpp

```diff
@@ -188,6 +188,19 @@ class Builder {
     item<Dims, false> LocalItem = createItem<Dims, false>(LocalSize, LocalId);
     return createNDItem<Dims>(GlobalItem, LocalItem, Group);
   }
+
+  template <int Dims, bool WithOffset>
+  static auto getElement(item<Dims, WithOffset> *)
+      -> decltype(getItem<Dims, WithOffset>()) {
+    return getItem<Dims, WithOffset>();
+  }
+
+  template <int Dims>
+  static auto getNDItem()
+      -> decltype(getElement(static_cast<nd_item<Dims> *>(nullptr))) {
```

Comment:
Idem

Reply:
The reasons are the same as in comment thread above.